src/sms-verification/scripts/directives.js

```diff
@@ -0,0 +1,64 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('user_management.sms-verification');
+
+    module.directive('smsVerify', [
+        '$parse',
+        '$routeParams',
+        'smsVerificationFactory',
+        function ($parse, $routeParams, smsVerificationFactory) {
+            return {
+                restrict: 'A',
+                scope: true,
+                templateUrl: 'templates/user_management/sms-verification/verify_form.html',
+                link: function (scope, element, attrs) {
+
+                    // 2-way bind data attribute to scope.
+                    // scope: { data: '=' }
+                    if (angular.isDefined(attrs.data)) {
+                        // Set scope.data form attrs.data.
+                        scope.$watch(attrs.data, function(data) {
```

Comment:
There's a function specifically for doing this with `attrs`. `attrs.$observe('data, function () {...});`

https://docs.angularjs.org/api/ng/type/$compile.directive.Attributes#$observe